docs/docs/alerting/notification-policies.md

```diff
@@ -9,7 +9,79 @@ For a complete explanation on notification policies, see the [upstream Grafana d
 {{% /alert %}}
 
+## Simple Notification Policy
 
 The following snippet shows an example notification policy routing to the `operations` or `security` team based on the `team` label.
 
 {{< readfile file="../examples/notification-policy/resources.yaml" code="true" lang="yaml" >}}
+
+## Dynamic Notification Policy Routes
+
+There might be scenarios where you can not define the entire notification policy in a single place and you have to assemble it from multiple resouces.
+In this case, you can use the `spec.route.routeSelector` in combination with multiple `GrafanaNotificationPolicyRoute` resources.
+
+The `routeSelector` can be specified in any `route` object, including the list of `spec.route.routes`.
+
+All `GrafanaNotificationPolicyRoute` resources will then be discovered based on the label selector defined in `spec.route.routeSelector`.
+In case `spec.allowCrossNamespaceImport` is enabled, matching routes will be fetched from all namespaces.
+Otherwise only routes from the same namespace as the `GrafanaNotificationPolicy` will be discovered.
+
+All discovered routes will then get set on the `spec.route.routes[]` of the `GrafanaNotificationPolicy`.
+
+{{% alert title="Note" color="secondary" %}}
+The `spec.route.routes` and `spec.route.routeSelector` fields are mutually exclusive.
+When both fields are specified, the `routeSelector` takes precedence and overrides anything defined in `routes`.
+{{% /alert %}}
+
+The following shows an example of how dynamic routes will get merged.
+
+{{< readfile file="../examples/notification-policy/routes.yaml" code="true" lang="yaml" >}}
+
+The resulting Notification Policy will be the following:
+
+```yaml
+apiVersion: 1
+policies:
+    - orgId: 1
+      receiver: grafana-default-email
+      group_by:
+        - grafana_folder
+        - alertname
+      routes:
+        - receiver: grafana-default-email
+          object_matchers:
+            - - inline
+              - =
+              - first
+            - - team
+              - =
+              - a
+          routes:
+            - receiver: grafana-default-email
+              object_matchers:
+                - - dynamic
+                  - =
+                  - e
+            - receiver: grafana-default-email
+              object_matchers:
+                - - crossNamespace
+                  - =
+                  - "true"
+                - - dynamic
+                  - =
+                  - c
+              routes:
+                - receiver: grafana-default-email
+                  object_matchers:
+                    - - dynamic
+                      - =
+                      - d
+        - receiver: grafana-default-email
+          object_matchers:
+            - - inline
+              - =
+              - second
+            - - team
+              - =
+              - b
+```
```

Comment:
I think a screenshot would be nicer here to avoid confusion between kubernetes resources and grafana exports